acmw-db-app/pages/signin.js

```diff
@@ -0,0 +1,129 @@
+import Head from 'next/head'
+import styles from '../styles/SignIn.module.css'
+import React from 'react';
+
+function SignIn() {
+  // TODO link to page that sends password by email?
+  return (
+    <div className={styles.container}>
+      <Head>
+        <title>ACM-W Database Sign In</title>
+      </Head>
+
+      <main className={styles.main}>
+        <div className={styles.card}>
+          <img src="/logo.png" alt="ACM-W Logo" className={styles.logo} />
+          <h1> OSU Database </h1>
+          <h2> Please sign in to continue </h2>
+          <TextInput
+            id="username"
+            label="Username"
+            predicted="Brutus Buckeye"
+          />
+          <div className={styles.passwordChunk}>
+            <TextInput
+              id="password"
+              label="Password"
+              type="password"
+            />
+            <a className={styles.smol} href="https://crouton.net/">Forgot username or password?</a>
+          </div>
+          <SignInButton
+          />
+        </div>
+      </main>
+    </div>
+  )
+}
+
+export default SignIn
+
+class TextInput extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      id: props.id,
+      active: false,
+      value: props.value || "",
+      error: props.error || "",
+      label: props.label || "",
+      type: props.type || "text",
+      predicted: props.predicted || "",
+    };
```

Comment:
This is probably fine but I don't know if you've heard about hooks to avoid state: https://reactjs.org/docs/hooks-intro.html

Basically what I was advised to do was avoid using classes and use functions w/hooks instead. Another article on that here (with better reasoning as to why at the end than I could come up with): https://medium.com/@Zwenza/functional-vs-class-components-in-react-231e3fbd7108

Reply:
Oh I didn't know about this; thanks! I will also try to change that soon.